\section{KEGG}
\label{sect:background_kegg}

\emph{The majority of this section is from ``The KEGG databases at GenomeNet'' by
Kanehisa et al \cite{kegg-basic}.}

\subsection{Overview}
\label{sect:background_kegg_overview}

The Kyoto Encyclopedia of Genes and Genomes (KEGG) is the primary database
resource of the \href{http://www.genome.ad.jp/}{Japanese GenomeNet
service}\footnote{\url{http://www.genome.ad.jp/}} for understanding higher order
functional meanings and utilities of the cell or the organism from its genome
information. KEGG consists of the PATHWAY database for the computerized
knowledge on molecular inter- action networks such as pathways and complexes,
the GENES database for the information about genes and proteins generated by
genome sequencing projects, and the LIGAND database for the information about
chemical compounds and chemical reactions that are relevant to cellular
processes. The KEGG databases are updated daily and made freely available at
\url{http://www.genome.ad.jp/kegg/}.

\subsection{PATHWAY Database}
\label{sect:background_kegg_pathway_database}

\subsubsection{Generalized protein interaction network}

The data object stored in the PATHWAY database is called the generalized protein
interaction network, or simply the network, which is a network of gene products
(nodes) with three types of interactions or relations (edges): enzyme–enzyme
relations which are two enzymes catalyzing successive reaction steps in the
metabolic pathway, direct protein–protein interactions such as binding and
phosphorylation, and gene expression relations involving transcription factors
and target gene products. The generalized protein interaction network is drawn
manually as a graphical pathway diagram (pathway map), and it is also stored as
a set of binary relations. The set of binary relations is a computable form of
the network information, but at the moment only the enzyme–enzyme relations are
maintained where a relation consists of a pair of nodes (enzymes) and an edge
(common compound) in between. As of September 7, 2001, the PATHWAY database
contains 5761 entries including 201 reference pathway diagrams and 83 ortholog
group tables, as well as 14 960 enzyme–enzyme relations. From the manually drawn
reference pathways, many organism-specific pathways are automatically generated
according to the ortholog identifier assignments in the GENES database. The
total number of gene product nodes that appear on the KEGG pathways is
approximately 6000, and roughly one-quarter to one-third of the genes in a
bacterial or archaeal genome can be mapped to one or more pathway diagrams. The
ortholog group tables contain the information about correlated clusters, which
are common subgraphs among multiple graphs. In this case a correlated cluster
represents a relationship between the positional correlation of genes in the
genome and the functional correlation of gene products in the network, such as a
set of genes in a conserved gene cluster (operon) forming a subpathway or a
complex. The total number of genes in the KEGG ortholog group tables is
approximately 26,000, which is about 10\% of the total number of genes in the
GENES database.

\subsubsection{Access methods}

The network information of the KEGG/PATHWAY database is hierarchically
categorized into four levels. According to the maintainers' view on the
hierarchy and modularity of cellular functions, the top level is categorized
into metabolism, genetic information processing, environmental information
processing, and the rest named cellular processes. In addition, a new top
category of human diseases is being introduced (see Table 2 for the top two
levels). The third level corresponds to a pathway diagram and/or an ortholog
group table, which is a collection of genes and proteins. The PATHWAY database
can best be viewed by following this hierarchy top-down in the KEGG table of
contents page where the top level item of metabolism is designated by
``Metabolic pathways'' and the rest of the top level items are designated by
``Regulatory pathways.'' Alternatively, the hierarchy may be used bottom-up
starting from the KEGG gene catalogs for individual organisms. In addition, the
text information describing PATHWAY entries can be searched by the DBGET/LinkDB
system.
